chainer/gradient_check.py

```diff
@@ -114,3 +115,146 @@ def assert_allclose(x, y, atol=1e-5, rtol=1e-4, verbose=True):
     except Exception:
         print('error:', numpy.abs(x - y).max())
         raise
+
+
+def _as_tuple(x):
+    if isinstance(x, tuple):
+        return x
+    elif isinstance(x, list):
+        return tuple(x)
+    else:
+        return (x,)
+
+
+def check_backward(func, x_data, y_grad, params=(),
+                   eps=1e-3, atol=1e-5, rtol=1e-4):
+    """Test backward procedure of a given function.
+
+    This function automatically check backward-process of given function.
+    For example, when you have a :class:`~chainer.Function` class ``MyFunc``,
+    that gets two arguments and returns one value, you can make its test like
+    this::
+
+    >> def test_my_func(self):
+    >>   func = MyFunc()
+    >>   x1_data = xp.array(...)
+    >>   x2_data = xp.array(...)
+    >>   gy_data = xp.array(...)
+    >>   check_backward(func, (x1_data, x2_data), gy_data)
+
+    This method creates :class:`~chainer.Variable` objects with ``x_data``
+    and calls ``func`` with the :class:`~chainer.Variable`s to get its result
+    as :class:`~chainer.Variable`.
+    Then, it sets ``y_grad`` arrays to ``grad`` attributed of the result and
```

Comment:
'Then, it sets ``y_grad`` array to ``grad`` attribute of the result' (remove *s* in 'arrays' and *d* in 'attributes')